crates/editor/src/editor.rs

```diff
@@ -1086,53 +1086,25 @@ impl CompletionsMenu {
 
         let completions = self.completions.read();
         matches.sort_unstable_by_key(|mat| {
-            // We do want to strike a balance here between what the language server tells us
-            // to sort by (the sort_text) and what are "obvious" good matches (i.e. when you type
-            // `Creat` and there is a local variable called `CreateComponent`).
-            // So what we do is: we bucket all matches into two buckets
-            // - Strong matches
-            // - Weak matches
-            // Strong matches are the ones with a high fuzzy-matcher score (the "obvious" matches)
-            // and the Weak matches are the rest.
-            //
-            // For the strong matches, we sort by the language-servers score first and for the weak
-            // matches, we prefer our fuzzy finder first.
-            //
-            // The thinking behind that: it's useless to take the sort_text the language-server gives
-            // us into account when it's obviously a bad match.
-
-            #[derive(PartialEq, Eq, PartialOrd, Ord)]
-            enum MatchScore<'a> {
-                Strong {
-                    sort_text: Option<&'a str>,
-                    score: Reverse<OrderedFloat<f64>>,
-                    sort_key: (usize, &'a str),
-                },
-                Weak {
-                    score: Reverse<OrderedFloat<f64>>,
-                    sort_text: Option<&'a str>,
-                    sort_key: (usize, &'a str),
-                },
-            }
-
             let completion = &completions[mat.candidate_id];
             let sort_key = completion.sort_key();
             let sort_text = completion.lsp_completion.sort_text.as_deref();
             let score = Reverse(OrderedFloat(mat.score));
-
-            if mat.score >= 0.2 {
-                MatchScore::Strong {
-                    sort_text,
-                    score,
-                    sort_key,
-                }
-            } else {
-                MatchScore::Weak {
-                    score,
-                    sort_text,
-                    sort_key,
-                }
-            }
+            let equals = match query.as_deref() {
+                Some(_) => !mat.string.eq_ignore_ascii_case(query.unwrap()),
+                None => true,
+            };
+            let matching_index = match query.as_deref() {
+                Some(_) => mat.string.to_lowercase().find(&query.unwrap().to_lowercase()).or(Some(usize::MAX)),
```

Comment:
I don't think this is a good spot to be using `to_lowercase` in. Since this is a comparator function for a sort, it can be invoked multiple times for a single element. We could either lowercase the keys beforehand or use something like `sort_by_cached_key`. I'd use the former if anything, as we don't need the sorting to be stable I guess?